virtcontainers/device/manager/manager.go

```diff
@@ -229,3 +230,39 @@ func (dm *deviceManager) IsDeviceAttached(id string) bool {
 	}
 	return d.GetAttachCount() > 0
 }
+
+// NewDevice creates a device based on specified DeviceInfo
+func (dm *deviceManager) LoadDevices(devStates []persistapi.DeviceState) {
+	dm.Lock()
+	defer dm.Unlock()
+
+	for _, ds := range devStates {
+		switch config.DeviceType(ds.Type) {
+		case config.DeviceGeneric:
+			dev := &drivers.GenericDevice{}
+			dev.Load(ds)
+			dm.devices[dev.DeviceID()] = dev
+		case config.DeviceBlock:
+			dev := &drivers.BlockDevice{}
+			dev.Load(ds)
+			dm.devices[dev.DeviceID()] = dev
+		case config.DeviceVFIO:
+			dev := &drivers.VFIODevice{}
+			dev.Load(ds)
+			dm.devices[dev.DeviceID()] = dev
+		case config.VhostUserSCSI:
+			dev := &drivers.VhostUserSCSIDevice{}
+			dev.Load(ds)
+			dm.devices[dev.DeviceID()] = dev
+		case config.VhostUserBlk:
+			dev := &drivers.VhostUserBlkDevice{}
+			dev.Load(ds)
+			dm.devices[dev.DeviceID()] = dev
+		case config.VhostUserNet:
+			dev := &drivers.VhostUserNetDevice{}
+			dev.Load(ds)
+			dm.devices[dev.DeviceID()] = dev
+		}
```

Comment:
How about adding a `default` case and logging a warning/error that the type cannot be handled?